entgql/internal/todopulid/types.go

```diff
@@ -0,0 +1,41 @@
+// Copyright 2019-present Facebook
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package todopulid
+
+import (
+	"context"
+	"fmt"
+
+	"github.com/facebookincubator/ent-contrib/entgql/internal/todopulid/ent/schema/pulid"
+	"github.com/facebookincubator/ent-contrib/entgql/internal/todopulid/ent/todo"
+)
+
+// TypeMap maps PULID prefixes to database table names.
+var TypeMap = map[pulid.ID]string{
+	"TO": todo.Table,
+}
```

Comment:
I actually expect those strings (e.g. `"TO"`) to be generated, or even generates this id2type map function. 
I can add this if you don't mind (I'll add you as a reviewer).

Reply:
Not at all! Would love the api insight -- also happy to implement given some direction.

Reply:
Sure. Did you figure out why CI is falling?

Reply:
> also happy to implement given some direction.

I thought about creating a `schema.Mixin` to be shared with all other schemas under the `pulid` package. For example:

```go

func MixinWithPrefix(prefix string) *Mixin {
	return &Mixin{prefix: prefix}
}

type Mixin struct {
	mixin.Schema
	prefix string
}

func (m Mixin) Fields() []ent.Field {
	return []ent.Field{
		field.String("id").
			GoType(ID("")).
			DefaultFunc(WithPrefix(m.prefix)),
	}
}

func (m Mixin) Annotations() []schema.Annotation {
	return []schema.Annotation{
		Annotation{Prefix: b.prefix},
	}
}
```

And then, we can use it as follows:

```go
// Mixin of the Todo.
func (Todo) Mixin() []ent.Mixin {
	return []ent.Mixin{
		pulid.MixinWithPrefix("TD"), // "TD" declared once.
	}
}
```

Also, define an external template that will generate the id-mapper from the mixed-in annotation:

```
// pulid/template/pulid.tmpl

var prefixMap = map[pulid.ID]string{
	{{- range $n := $.Nodes }}
		{{- if $annotation := $f.Annotations.PULID }}
			"{{ $annotation.Prefix }}": {{ $n.Package }}.Table,
		{{- end }}
	{{- end }}
}

// Exported as `ent.IDToType(...)`.
func IDToType(ctx context.Context, id pulid.ID) (string, error) {
	// ...
}
```

WDYT?

Reply:
Love that direction. I’ll flesh out the idea a bit more.